src/scene_battle_rpg2k3.cpp

```diff
@@ -2293,15 +2438,38 @@ Scene_Battle_Rpg2k3::BattleActionReturn Scene_Battle_Rpg2k3::ProcessBattleAction
 		return BattleActionReturn::eContinue;
 	}
 
-	SetWait(30, 30);
-
 	// If action does multiple attacks, repeat again.
 	if (action->RepeatNext(false)) {
 		SetBattleActionState(BattleActionState_StartAlgo);
 		return BattleActionReturn::eContinue;
 	}
 
-	SetBattleActionState(BattleActionState_PostAction);
+	if (source->GetType() == Game_Battler::Type_Ally) {
+		if (action->GetType() == Game_BattleAlgorithm::Type::Normal) {
+			auto* actor = static_cast<Game_Actor*>(source);
+			auto* source_sprite = actor->GetActorBattleSprite();
+			if (source_sprite) {
+				source_sprite->SetNormalAttacking(false);
+				auto* weapon = actor->GetWeaponSprite();
+				int weapon_animation_id = 0;
+				if (weapon) {
+					auto* weapon_animation_data = action->GetWeaponAnimationData();
+					if (weapon_animation_data) {
+						if (weapon_animation_data->type == 0) {
```

Comment:
can you check if there is a enum for this 0 somewhere in liblcf?

Reply:
Fixed. A enum exists for this one.